File: 윤석주/day1/크레인인형뽑기.py
def solution(board, moves):
    answer = 0
    result = []
    numberOfRaw = len(board)
    for move in moves:
        current = 0
        for raw in range(numberOfRaw):
            if board[raw][move-1] != 0:
                current = board[raw][move-1]
                board[raw][move-1] = 0
                break
        if current != 0:
            if result:
                if result[-1] == current:
                    result.pop()
                    answer += 2
                else:
                    result.append(current)
            else:
                result.append(current)
    return answer

File: 한진영/크레인인형뽑기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    vector<int> doll;
    int answer = 0;
    for(int i=0; i<moves.size(); i++){
        int col = moves[i]-1;
        for(int row=0; row<board[0].size(); row++){
            if(board[row][col]){
                doll.push_back(board[row][col]);
                board[row][col]=0;
                if(doll.size()>1 && doll[doll.size()-1] == doll[doll.size()-2]){
                    doll.pop_back();
                    doll.pop_back();
                    answer+=2;
                }
                break;
            }
        }
    }
    return answer;
}